transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1243,6 +1244,8 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
 
         @Override
         public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+            logger.debug("Reached end of the pipeline for handlers : {}.",
+                    Arrays.toString(ctx.pipeline().names().toArray()));
```

Comment:
`Arrays.toString()` and `toArray()` is not needed. Also will be better with check for logger level bcz `.names()` is not free:
```java
if (logger.isDebugEnabled()) {
    logger.debug("Reached end of the pipeline for handlers : {}.", ctx.pipeline().names());
}
```

Reply:
@fenik17 good points. Let me fix.